src/PIL/ImageFile.py

```diff
@@ -322,7 +322,7 @@ def load(self) -> Image.core.PixelAccess | None:
 
     def load_prepare(self) -> None:
         # create image memory if necessary
-        if self._im is None or self.im.mode != self.mode or self.im.size != self.size:
```

Comment:
Not sure now it affects multipage images with different page sizes

Reply:
When seeking to a different TIFF frame, `self.im` is populated afresh.

https://github.com/python-pillow/Pillow/blob/08d9c89d8a7db84e8145814bca36d7dd5b7465ce/src/PIL/TiffImagePlugin.py#L1189-L1198

Reply:
Yeah, I saw this and tried to remove this initialization and relay on `load_prepare` instead, for some reasons it didn't work.

For me providing correct canvas (self.im) in the base class is more preferable than recreating it the plugin. There was [another approach](https://github.com/uploadcare/pillow-simd/commit/1d77bfe9d939e208e3bb2b897a4209c0881eb9f1#diff-6ad43f85f1a075181d4d8cfcd97ae27bba1eccf5c3db5a3457160f98218eba6eR1205) which doesn't introduce difference between `im.size` and `im._size`.

Reply:
> For me providing correct canvas (self.im) in the base class is more preferable than recreating it in the plugin.

If you think I'm creating a core image instance an extra time in `TiffImagePlugin.load_prepare()` at the moment, that's not the case - the creating of the core image in `ImageFile.load_prepare()` is skipped [because](https://github.com/python-pillow/Pillow/blob/e80cada0f691ef60d4448145dc9dd0319c13b68f/src/PIL/ImageFile.py#L323-L326) it's already performed here.

Reply:
> Yeah, I saw this and tried to remove this initialization and relay on load_prepare instead, for some reasons it didn't work.

I've pushed https://github.com/python-pillow/Pillow/pull/8390/commits/3ab41ec7dbf55ea535d1691a102fa17fe9bfa4b5, see what you think.

Reply:
Ah, I see now - my `load_prepare()` would create a new core image when the image was loaded after `seek()`.

I've pushed a commit following your other approach.